include/cinder/FileWatcher.h

```diff
@@ -0,0 +1,157 @@
+	void	setConnectToAppUpdateEnabled( bool enable );
+	//! Returns whether file watching is enabled or disabled.
+	bool	isConnectToAppUpdateEnabled() const		{ return mConnectToAppUpdateEnabled; }
+
+	//! Adds a single file at \a filePath to the watch list. Does not immediately call \a callback, but calls it whenever the file has been updated.
+	signals::Connection watch( const fs::path &filePath, const std::function<void ( const WatchEvent& )> &callback );
+	//! Adds a single file at \a filePath to the watch list, with optional \a options.
+	signals::Connection watch( const fs::path &filePath, const Options &options, const std::function<void ( const WatchEvent& )> &callback );
+	//! Adds the files in \a filePaths to the watch list. Does not immediately call \a callback, but calls it whenever one of the files has been updated.
+	signals::Connection watch( const std::vector<fs::path> &filePaths, const std::function<void ( const WatchEvent& )> &callback );
+	//! Adds the files in \a filePaths to the watch list, with optional \a options.
+	signals::Connection watch( const std::vector<fs::path> &filePaths, const Options &options, const std::function<void ( const WatchEvent& )> &callback );
+
+	//! Removes any watches for \a filePath
+	void	unwatch( const fs::path &filePath );
+	//! Removes any watches for \a filePaths
+	void	unwatch( const std::vector<fs::path> &filePaths );
+	//! Marks a file already in the watch list as active.
+	void	enable( const fs::path &filePath );
+	//! Marks a file as inactive, but doesn't remove it from the watch list.
+	void	disable( const fs::path &filePath );
+
+	//! Processes any WatchEvents with modified files that need callbacks. By default, this is called automatically from from the App's update signal, if there is currently an App instance.
+	//! Callbacks will occur on the thread that this method is called from. If you wish to call this manually, first call `setConnectToAppUpdateEnabled( false )`.
+	void	update();
+
+	//! Returns the number of Watch instances being watched.
+	const size_t	getNumWatches() const	{ return mWatchList.size(); }
+	//! Returns the total number of watched files, taking into account the number of files being watched by a WatchMany
+	const size_t	getNumWatchedFiles() const;
+
+	//! Sets the update time interval in seconds for the polling thread. \default is 0.02 seconds.
+	void		setThreadUpdateInterval( double seconds )	{ mThreadUpdateInterval = seconds; }
+	//! Returns the update time interval in seconds for the polling thread. \default is 0.02 seconds.
+	double		getThreadUpdateInterval() const				{ return mThreadUpdateInterval; }
+
+  private:
+	FileWatcher();
+
+	void	configureWatchPolling();
+	void	stopWatchPolling();
+	void	threadEntry();
+
+	std::list<std::unique_ptr<Watch>>	mWatchList;
+	std::recursive_mutex				mMutex;
+	std::unique_ptr<std::thread>		mThread;
+	std::atomic<bool>					mThreadShouldQuit;
+	std::atomic<double>					mThreadUpdateInterval		= { 0.02 };
```

Comment:
Currently we use doubles to represent time throughout the codebase. This is also a private variable so I think double is fine here, and more precision than it'd ever need.